mindsdb/integrations/handlers/leonardoai_handler/leonardo_ai_handler.py

```diff
@@ -0,0 +1,234 @@
+import os
+import requests
+import time
+from typing import Dict, Optional
+from mindsdb.integrations.libs.llm_utils import get_completed_prompts
+
+import pandas as pd
+
+from mindsdb.integrations.libs.base import BaseMLEngine
+from mindsdb.utilities import log
+from mindsdb.interfaces.storage.model_fs import HandlerStorage
+from mindsdb.utilities.config import Config
+
+logger = log.getLogger(__name__)
+
+LEONARDO_API_BASE = 'https://cloud.leonardo.ai/api/rest/v1'
+
+
+class LeonardoAIHandler(BaseMLEngine):
+    """
+    This integration seamlessly combines MindsDB and Leonardo AI to create a powerful
+    AI-driven solution for creative content generation.
+
+    Content Generation with Leonardo AI: Harness the power of advanced generative
+    models for creative content production. From realistic images to artistic text, Leonardo
+    AI opens up new possibilities for content creators.
+    """
+    name = "leonardoai"
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.all_models = []
+        self.default_model = '6bef9f1b-29cb-40c7-b9df-32b51c1f67d3'
+        self.base_api = LEONARDO_API_BASE
+        self.rate_limit = 50
+        self.max_batch_size = 5  # default value
+
+    def create(self, target: str, args=None, **kwargs):
+
+        args['target'] = target
+
+        if "using" not in args:
+            raise Exception(
+                "Leornardo Engine requires a USING clause!"
+            )
+
+        self.model_storage.json_set("args", args)
+        api_key = self._get_leonardo_api_key(args, self.engine_storage)  # fetch api key
+
+        # check if API key is valid
+        self.connection = requests.get(
+            "https://cloud.leonardo.ai/api/rest/v1/me",
+            headers={
+                "accept": "application/json",
+                "authorization": f"Bearer {api_key}"
+            }
+        )
+
+        # if valid, check if the model is valid
+        if(self.connection.status_code == 200):
+            # get all the available models
+            available_models = self._get_platform_model(args)
+
+            if not args['using']['model']:
+                args['using']['model'] = self.default_model
+            elif args['using']['model'] not in available_models:    # if invalid model_id is provided
+                raise Exception("Invalid Model ID. Please use a valid Model")
+
+        # API key is invalid
+        else:
+            raise Exception("Unable to make connection, please verify the API key.")
```

Comment:
Maybe wrap the above in try/except and return Auth exception

Reply:
I wrapped the if/else within try/except block.